server/limits_test.go

```diff
@@ -0,0 +1,313 @@
+			mock.Anything, mock.Anything)
+
+		err := p.sendConcurrentSessionsWarning()
+		require.NoError(t, err)
+	})
+
+	t.Run("team", func(t *testing.T) {
+		mockAPI.On("GetLicense").Return(nil).Once()
+
+		mockAPI.On("GetUsers", mock.AnythingOfType("*model.UserGetOptions")).Return([]*model.User{
+			{
+				Id:     "adminID",
+				Locale: "it",
+			},
+		}, nil).Once()
+
+		mockAPI.On("GetDirectChannel", "adminID", "botID").Return(&model.Channel{
+			Id: "channelID",
+		}, nil).Once()
+
+		mockAPI.On("IsEnterpriseReady").Return(false).Once()
+
+		mockAPI.On("CreatePost", &model.Post{
+			UserId:    "botID",
+			ChannelId: "channelID",
+			Message:   ":warning: app.admin.concurrent_sessions_warning.intro\r\n\r\napp.admin.concurrent_sessions_warning.team",
+		}).Return(&model.Post{Id: "postID"}, nil).Once()
+
+		err := p.sendConcurrentSessionsWarning()
+		require.NoError(t, err)
+	})
+
+	t.Run("e0", func(t *testing.T) {
+		mockAPI.On("GetLicense").Return(nil).Once()
+
+		mockAPI.On("GetUsers", mock.AnythingOfType("*model.UserGetOptions")).Return([]*model.User{
+			{
+				Id:     "adminID",
+				Locale: "it",
+			},
+		}, nil).Once()
+
+		mockAPI.On("GetDirectChannel", "adminID", "botID").Return(&model.Channel{
+			Id: "channelID",
+		}, nil).Once()
+
+		mockAPI.On("IsEnterpriseReady").Return(true).Once()
+
+		mockAPI.On("CreatePost", &model.Post{
+			UserId:    "botID",
+			ChannelId: "channelID",
+			Message:   ":warning: app.admin.concurrent_sessions_warning.intro\r\n\r\napp.admin.concurrent_sessions_warning.pro_or_e0",
+		}).Return(&model.Post{Id: "postID"}, nil).Once()
+
+		err := p.sendConcurrentSessionsWarning()
+		require.NoError(t, err)
+	})
+
+	t.Run("professional", func(t *testing.T) {
+		mockAPI.On("GetLicense").Return(&model.License{
+			SkuShortName: model.LicenseShortSkuProfessional,
+		}, nil).Once()
+
+		mockAPI.On("GetUsers", mock.AnythingOfType("*model.UserGetOptions")).Return([]*model.User{
+			{
+				Id:     "adminID",
+				Locale: "it",
+			},
+		}, nil).Once()
+
+		mockAPI.On("GetDirectChannel", "adminID", "botID").Return(&model.Channel{
+			Id: "channelID",
+		}, nil).Once()
+
+		mockAPI.On("CreatePost", &model.Post{
+			UserId:    "botID",
+			ChannelId: "channelID",
+			Message:   ":warning: app.admin.concurrent_sessions_warning.intro\r\n\r\napp.admin.concurrent_sessions_warning.pro_or_e0",
+		}).Return(&model.Post{Id: "postID"}, nil).Once()
+
+		err := p.sendConcurrentSessionsWarning()
+		require.NoError(t, err)
+	})
+
+	t.Run("enterprise", func(t *testing.T) {
+		mockAPI.On("GetLicense").Return(&model.License{
+			SkuShortName: model.LicenseShortSkuEnterprise,
+		}, nil).Once()
+
+		mockAPI.On("GetUsers", mock.AnythingOfType("*model.UserGetOptions")).Return([]*model.User{
+			{
+				Id:     "adminID",
+				Locale: "it",
+			},
+		}, nil).Once()
+
+		mockAPI.On("GetDirectChannel", "adminID", "botID").Return(&model.Channel{
+			Id: "channelID",
+		}, nil).Once()
+
+		mockAPI.On("CreatePost", &model.Post{
```

Comment:
I wonder if it would be good to use `mockAPI.AssertNumberOfCalls...` for these, since that's what you're testing, right?

Reply:
@cpoile I'm not sure I fully understand what you mean by *these*.

In general, I try to be as specific as possible so we can verify the inputs and outputs of our mocked methods rather than just whether they were called.

Reply:
What I mean is: keep what you have, and then for the API calls that we are assuming are being called (esp. the `CreatePost`), do a `mockAPI.AssertNumberOfCalls(t, "CreatePost", 1)` to make sure. That's the real test, right?

Reply:
@cpoile Ah. Are you saying that if `CreatePost` doesn't execute for some reason, it won't complain?

Reply:
> @cpoile Ah. Are you saying that if `CreatePost` doesn't execute for some reason, it won't complain?

Yah, iirc once just says mock this once, but doesn't complain if it doesn't get called at all.

Reply:
I'll fix the ones touched in here, but then I'll need to send a separate PR to review the rest.